### PR TITLE
canbus: canopen: calculate flash image crc without holding can od lock and read from flash in 32 byte chunks

### PR DESCRIPTION
**Pull request by ksorensen**
Flash image crc is calculated when reading can index 0x1f56.

- Release can od lock during image crc calculation, since this can take upwards of 100ms or more
- Read 32 bytes at a time from flash instead of 4 bytes at a time, to reduce flash reading overhead

### CODE REVIEW

File: subsys/canbus/canopen/canopen_program.c [hunk 1]
```diff
@@ -276,15 +276,48 @@ static CO_SDO_abortCode_t canopen_odf_1f51(CO_ODF_arg_t *odf_arg)
 }
 
 #ifdef CONFIG_BOOTLOADER_MCUBOOT
+/** Calculate crc for region in flash
```
Comment (henrikbrixandersen) [inline subsys/canbus/canopen/canopen_program.c:279]:
```suggestion
/** @brief Calculate crc for region in flash
```

File: subsys/canbus/canopen/canopen_program.c [hunk 2]
```diff
@@ -276,15 +276,48 @@ static CO_SDO_abortCode_t canopen_odf_1f51(CO_ODF_arg_t *odf_arg)
...
+ *
+ * @param flash_area Flash area to read from, must be open
+ * @offset Offset to read from
+ * @size Number of bytes to include in calculation
+ * @pcrc Pointer to uint32_t where crc will be written if return value is 0
+ *
+ * @return 0 if successful
```
Comment (henrikbrixandersen) [inline subsys/canbus/canopen/canopen_program.c:286]:
```suggestion
 * @return 0 if successful, negative errno on failure
```

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

File: subsys/canbus/canopen/canopen_program.c (hunk 1 above)
Comment (ksorensen) [reply subsys/canbus/canopen/canopen_program.c]: Fixed.

File: subsys/canbus/canopen/canopen_program.c (hunk 2 above)
Comment (ksorensen) [reply subsys/canbus/canopen/canopen_program.c]: Added as suggested.

**Review by henrikbrixandersen — APPROVED** [review]

**Review by alexanderwachter — APPROVED** [review]

**Pull request merged**